File: two-pointer/11-Container-With-Most-Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        ans = 0
        left, right = 0, len(height) - 1
        while left < right:
            minHeight = min(height[left], height[right])
            current_container = minHeight * (right - left)
            ans = max(ans, current_container)
            left += 1
            right -= 1
        return ans

File: reference/two-pointer/11-Container-With-Most-Water.go
func maxArea(height []int) int {
    ans := 0
    left, right := 0, len(height) - 1
    for left < right {
        minHeight := min(height[left], height[right])
        currentContainer := minHeight * (right - left)
        if currentContainer > ans {
            ans = currentContainer
        }

        if height[left] < height[right] {
            left++
        } else {
            right--
        }
    }
    return ans
}

func min(a int, b int) int {
    if a < b {
        return a
    }
    return b
}